File: Doctor/WebContent/Components/Doctor.js
$(document).ready(function() {
	if ($("#alertSuccess").text().trim() == "") {
		$("#alertSuccess").hide();
	}
	$("#alertError").hide();
});
// SAVE ============================================
$(document).on("click", "#btnSave", function(event) {
	// Clear alerts---------------------
	$("#alertSuccess").text("");
	$("#alertSuccess").hide();
	$("#alertError").text("");
	$("#alertError").hide();
	// Form validation-------------------
	var status = validateDoctorForm();
	if (status != true) {
		$("#alertError").text(status);
		$("#alertError").show();
		return;
	}
	// If valid------------------------
	
	var method = ($("#didDoctorIDSave").val() == "") ? "POST" : "PUT";

	$.ajax({
		url : "DoctorAPI",
		type : method,
		data : $("#formDoctor").serialize(),
		dataType : "text",
		complete : function(response, status) {
			onDoctorSaveComplete(response.responseText, status);
		}
	});
	
	//$("#formDoctor").submit();
});
// UPDATE==========================================
$(document).on("click",".btnUpdate",function(event) {
			$("#didDoctorIDSave").val($(this).closest("tr").find('#didDoctorIDUpdate').val());
			$("#docName").val($(this).closest("tr").find('td:eq(0)').text());
			$("#docAge").val($(this).closest("tr").find('td:eq(1)').text());
			$("#docGender").val($(this).closest("tr").find('td:eq(2)').text());
			$("#docSpecialization").val($(this).closest("tr").find('td:eq(3)').text());
			$("#hosID").val($(this).closest("tr").find('td:eq(4)').text());
			
		});

//---------------------------------------------------------------------------------------------------lol

function onDoctorSaveComplete(response, status)
{
    if (status == "success")
    {
        var resultSet = JSON.parse(response);
        if (resultSet.status.trim() == "success")
        {
            $("#alertSuccess").text("Successfully saved.");
            $("#alertSuccess").show();
            $("#divItemsGrid").html(resultSet.data);
        } 
        else if (resultSet.status.trim() == "error")
        {
            $("#alertError").text(resultSet.data);
            $("#alertError").show();
        }
    } 
    else if (status == "error")
    {
        $("#alertError").text("Error while saving.");
        $("#alertError").show();
    } 
    else
    {
        $("#alertError").text("Unknown error while saving..");
        $("#alertError").show();
    }
    
    $("#didDoctorIDSave").val("");
    $("#formDoctor")[0].reset();
}

$(document).on("click", ".btnRemove", function(event)
{
    $.ajax(
    {
        url : "DoctorAPI",
        type : "DELETE",
        data : "docID=" + $(this).data("doctorid"),
        dataType : "text",
        complete : function(response, status)
        {
            onDoctorDeleteComplete(response.responseText, status);
        }
    });
});

function onDoctorDeleteComplete(response, status)
{
    if (status == "success")
    {
        var resultSet = JSON.parse(response);
        
        if (resultSet.status.trim() == "success")
        {
            $("#alertSuccess").text("Successfully deleted.");
            $("#alertSuccess").show();
            $("#divItemsGrid").html(resultSet.data);
        } 
        else if (resultSet.status.trim() == "error")
        {
            $("#alertError").text(resultSet.data);
            $("#alertError").show();
        }
    } 
    else if (status == "error")
    {
        $("#alertError").text("Error while deleting.");
        $("#alertError").show();
    } 
    else
    {
        $("#alertError").text("Unknown error while deleting..");
        $("#alertError").show();
    }
}

//----------------------------------------------------------------------------------------------lol
// CLIENTMODEL=========================================================================
function validateDoctorForm() {
	// CODE
	if ($("#docName").val().trim() == "") {
		return "Insert Doctor name.";
	}
	// NAME
	if ($("#docAge").val().trim() == "") {
		return "Insert Doctor Age.";
	}
	// PRICE-------------------------------

	if ($("#docGender").val().trim() == "") {
		return "Insert Doctor Gender.";
	}
	
	if ($("#docSpecialization").val().trim() == "") {
		return "Insert Doctor Specialization.";
	}
	
	if ($("#hosID").val().trim() == "") {
		return "Insert Hospital ID.";
	}

	return true;
}

File: Doctor/src/model/Doctor.java
package model;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.Statement;
import java.sql.DriverManager;
import database.dbconnect;

public class Doctor {
	
	dbconnect obj=new dbconnect();
	
	public String insertDoctor(String docName, String docAge, String docGender, String docSpecialization,String hosID) {
				
				String output = "";
				try {
					Connection con = obj.connect();
					if (con == null) {
						return "Error while connecting to the database for inserting.";
					}
					// create a prepared statement
					
					 String query = " insert into doctor(`docName`,`docAge`,`docGender`,`docSpecialization`,`hosID`)" +
				                " values (?, ?, ?, ?, ?)";
					PreparedStatement preparedStmt = con.prepareStatement(query);
					
					preparedStmt.setString(1, docName);
					preparedStmt.setString(2, docAge);
					preparedStmt.setString(3, docGender);
					preparedStmt.setString(4, docSpecialization);
					preparedStmt.setString(5, hosID);
		
					preparedStmt.execute();
					con.close();
					
					 String newDoctor = readDoctor(); 
			            output = "{\"status\":\"success\",\"data\": \"" +newDoctor + "\"}";
					
					//output = "Inserted successfully";
					System.out.println("Inserted successfully.......................................");
				} catch (Exception e) {
					
					 output = "{\"status\":\"error\", \"data\":\"Error while inserting the Doctors.\"}";
		                System.err.println(e.getMessage());
		                
					//output = "Error while inserting the Doctors.";
					//System.out.println("Error while inserting the Doctors........."+ e);
					//System.err.println(e.getMessage());
				}
				return output;
			}
	 //
			public String readDoctor() {
				String output = "";
				try {
					Connection con = obj.connect();
					if (con == null) {
						return "Error while connecting to the database for reading.";
					}
					// Prepare the html table to be displayed
					//output = "<table border='1'>"
							//+ "<th>Doctor Name</th>"
							//+ "<th>Doctor Age</th>"
							//+ "<th>Doctor Gender</th>"
							//+ "<th>Doctor Specialization</th>"
							//+ "<th>Hospital ID</th>"
							//+ "<th>Update</th>"
							//+ "<th>Remove</th></tr>";
					
					output ="<table border='1'><tr><th>Doctor Name</th><th>Doctor Age</th><th>"
							+ "Doctor Gender</th><th>Doctor Specialization</th><th>Hospital ID</th><th>Update</th><th>Remove</th></tr>";
					String query = "select * from doctor";
					Statement stmt = con.createStatement();
					ResultSet rs = stmt.executeQuery(query);
					// iterate through the rows in the result set
					while (rs.next()) {
						String docID = Integer.toString(rs.getInt("docID"));
						String docName = rs.getString("docName");
						String docAge = rs.getString("docAge");
						String docGender = rs.getString("docGender");
						String docSpecialization = rs.getString("docSpecialization");
						String hosID = rs.getString("hosID");
						
						
						
						
						// Add into the html table
						/*output += "<tr><td><input id=\"didDoctorIDUpdate\"name=\"didDoctorIDUpdate\"type=\"hidden\" value=\"" + docID + "\">"
	                            + docName + "</td>";
						output += "<td>" + docAge + "</td>";
						output += "<td>" + docGender + "</td>";
						output += "<td>" + docSpecialization + "</td>";
						output += "<td>" + hosID + "</td>";*/
						
						output += "<tr><td><input id='didDoctorIDUpdate'name='didDoctorIDUpdate' type='hidden'value='" + docID + "'>" + 
						docName  + "</td>";
						output += "<td>" + docAge + "</td>";
						output += "<td>" + docGender + "</td>";
						output += "<td>" + docSpecialization + "</td>";
						output += "<td>" + hosID + "</td>";
						
					
						
						// buttons
						/*output += "<td><input name=\"btnUpdate\" type=\"submit\"value=\"Update\" class=\"btn btn-warning btnUpdate\"></td>"
								+ "<td><form method=\"post\" action=\"Doctor_Insert.jsp\">"
								+ "<input name=\"btnRemove\" type=\"submit\" value=\"Remove\"class=\"btn btn-danger\">"
								+ "<input name=\"didDoctorIDDelete\" type=\"hidden\" value=\"" + docID + "\">" + "</form></td></tr>";*/
						
						output += "<td><input name='btnUpdate' type='button'"
						+ "value='Update'"+"class='btnUpdate btn btn-secondary'></td>"
						+"<td><input name='btnRemove' type='button'"
						+"value='Remove'"+"class='btnRemove btn btn-danger' data-doctorid='"+ docID + "'>" + "</td></tr>";
						//1233
					}
					con.close();
					// Complete the html table
					output += "</table>";
				} catch (Exception e) {
					output = "Error while reading the Doctors.";
					System.err.println(e.getMessage());
				}
				return output;
			}
			
			public String updateDoctor(String docID, String docName, String docAge , String docGender , String docSpecialization,String hosID) {
				System.out.println("Update method...............................................................................");
				String output = "";
				try {
					Connection con = obj.connect();
					if (con == null) {
						return "Error while connecting to the database for updating.";
					}
					//update
					// create a prepared statement
					String query = "UPDATE doctor SET docName=?,docAge=?,docGender=?,docSpecialization=?,hosID=? WHERE docID=?";
					PreparedStatement preparedStmt = con.prepareStatement(query);
					// binding values
					preparedStmt.setString(1, docName);
					preparedStmt.setString(2, docAge);
					preparedStmt.setString(3, docGender);
					preparedStmt.setString(4, docSpecialization);
					preparedStmt.setString(5, hosID);
					preparedStmt.setInt(6, Integer.parseInt(docID));
					// execute the statement
					preparedStmt.execute();
					con.close();
					
					 String newDoctor = readDoctor(); 
			            
			            output = "{\"status\":\"success\", \"data\": \"" +newDoctor + "\"}";
					//output = "Updated successfully";
				} catch (Exception e) {
					 output = "{\"status\":\"error\", \"data\":\"Error while updating the Doctors.\"}";
					//output = "Error while updating the Doctor.";
					System.err.println(e.getMessage());
				}
				return output;
			}
			
			public String deleteDoctor(String docID) {
				String output = "";
				try {
					Connection con = obj.connect();
					if (con == null) {
						return "Error while connecting to the database for deleting.";
					}
					// create a prepared statement
					String query = "delete from doctor where docID=?";
					PreparedStatement preparedStmt = con.prepareStatement(query);
					// binding values
					preparedStmt.setInt(1, Integer.parseInt(docID));
					// execute the statement
					preparedStmt.execute();
					con.close();
					 String newDoctor = readDoctor();
			            output = "{\"status\":\"success\", \"data\": \"" +newDoctor + "\"}";
					//output = " Doctor Deleted successfully";
				} catch (Exception e) {
					
					output = "{\"status\":\"error\", \"data\":\"Error while deleting the Doctor.\"}"; 
					//output = "Error while deleting the Doctor.";
					System.err.println(e.getMessage());
				}
				return output;
			}
			
}
